### ISSUE TITLE
Replace Post/Block tab switcher with `TabPanel`

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#43414]
Prerequisite for #41904 #42320

The Post/Block tab switcher in the Inspector (`.edit-post-sidebar__panel-tabs`) is currently implemented as a list of buttons, and is missing `tabpanel` semantics.

<img width="285" alt="The Post/Block tab switcher" src="https://user-images.githubusercontent.com/555336/185593707-8bc4570a-7e41-4e1f-92e8-aed493d2330c.png">

This should be replaced with the TabPanel component.

**Comment by talldan** [comment]
I found this old PR that attempted to do this, but was closed - [Use TabPanel in edit settings sidebar header (#13587)](https://github.com/WordPress/gutenberg/pull/16663#top).

It looks like one issue surfaced in that PR (and has come up a few times recently in other issues) is that `TabPanel` may have to become a controlled component. I think when you deselect a block the selected tab changes from 'Block' to 'Post'.

**Comment by mirka** [comment]
> It looks like one issue surfaced in that PR (and has come up a few times recently in other issues) is that `TabPanel` may have to become a controlled component.

Ah, #46704 may unblock it then.

**Comment by alexstine** [comment]
@aaronrobertshaw asked for my feedback over here in another issue.

I highly support this now that the `TabPanel` is working better than it ever has before. Cannot wait to see this happen. This will move us another step in the right direction of having consistent controls/interactions.

Thanks.

**Comment by mirka** [comment]
Closed by #55360
